hie-compat/src-ghc901/Compat/HieAst.hs

```diff
@@ -0,0 +1,2031 @@
+{-
+Forked from GHC v9.0.1 to work around the readFile side effect in mkHiefile
```

Comment:
I also don't think we need to duplicate this. The "side effect free" `mkHieFile` is called `mkHieFileWithSource` in the GHC 9.0 module. A rename/re-export should suffice.

Reply:
Ok, that's great

Reply:
I've removed everything but the modified version of `mkHieFile`, which calls `mkHieFileWithSource` internally now.